samples/bluetooth/mesh/dfu/common/src/dfu_target.c

```diff
@@ -250,7 +259,13 @@ static void image_version_load(void)
 		return;
 	}
 
-	*img_ver = img_header.h.v1.sem_ver;
+	img_ver = &img_header.h.v1.sem_ver;
+
+	fwid.company_id = sys_cpu_to_le16(CONFIG_BT_COMPANY_ID);
+	fwid.major = img_ver->major;
+	fwid.minor = img_ver->minor;
+	fwid.revision = sys_cpu_to_le16(img_ver->revision);
+	fwid.build_num = sys_cpu_to_le32(img_ver->build_num);
```

Comment:
Wait ... I don't think endianness conversion to LE is needed. Isn't `mcuboot_img_sem_ver` has exact same internal structure like `fwid`?

Reply:
`mcuboot_img_sem_ver` uses system endianness. The previous code which just dumped `mcuboot_img_sem_ver` into the FWID was incorrect with regards to endianness.

Reply:
Really? Is it a hidden knowledge? API does not say anything like that.

Reply:
What? mcuboot_img_sem_ver is just a struct of integers. Integers are system-endian.

Reply:
Do note, I tested this, the current code has the correct behavior (i.e. `img_ver->revision == 0x1234` causes the Firmware Update Information Status::Current Firmware ID to contain "...3412...").

Reply:
Nevermind, the macro takes care of this, effectively nothing is changed by invoking the `sys_cpu_to_le16` function :).